### DISCUSSION TITLE
Future of this project

### DISCUSSION THREAD

**Discussion by leviathan747** [rrevenantt/antlr4rust#71] [General]
I love this project by the way, good work!

I was wondering what the outlook is for ever getting the rust target merged into the main branch of the antlr tool and also getting the runtime library to a stable "v1.0". I have a project currently written in Java which relies on an antlr parser and I would like to port it to rust and keep the same antlr grammar (and mostly make a direct port of the visitor).

I'm a bit of a rust novice, but I would be interested in understanding the roadmap to get there and potentially investing my time to get it there. I think this has a lot of promise but I can't reasonably use it in production until the target is in the antlr main line.

**Comment by Luna-Klatzer** [comment]
I was wondering the same and would hope that this project eventually gets to a stable release. Especially, since I'd also love to use the Rust target for WebAssembly compilations.

**Comment by Code7R** [comment]
Similar situation here, I am struggling with the setup of this project. The README tells us to use version 0.3 (and older are not supported) but there is no version 0.3, only some notes about 0.3-prerelease. Not much happening in https://github.com/rrevenantt/antlr4/tree/rust-target either.

Looks like the author is MIA (who knows, mind the "special operation" in UKR). Time to fork, anyone?

**Comment by khoshrou** [comment]
Guys, what's the state of this project?

**Comment by rrevenantt** [comment]
See antlr/antlr4#1839

вт, 14 окт. 2025 г. в 20:27, R.M Khoshrou ***@***.***>:
